cmd/atlas/internal/cmdapi/schema.go

```diff
@@ -611,6 +611,8 @@ func computeDiff(ctx context.Context, differ *sqlclient.Client, from, to *stateR
 	}
 	var diff []schema.Change
 	switch {
+	case to.hcl && len(desired.Schemas) > 0 && desired.Schemas[0].Name != from.schema:
+		return nil, fmt.Errorf("schema mismatch: %q <> %q", from.schema, desired.Schemas[0].Name)
```

Comment:
```suggestion
	// In case an HCL file is compared against a specific database schema (not a realm).
	case to.hcl && len(desired.Schemas) > 0 && desired.Schemas[0].Name != from.schema:
		return nil, fmt.Errorf("mismatched HCL and database schemas: %q <> %q", from.schema, desired.Schemas[0].Name)
```